### ISSUE TITLE
Crashes when tries to go for location settings

### ISSUE THREAD

**Issue by asadwaheed1** [BirjuVachhani/locus-android#77]
Device: Poco F3
OS: Android 12

```
java.lang.IncompatibleClassChangeError: Found interface com.google.android.gms.location.SettingsClient, but class was expected (declaration of 'com.google.android.gms.location.SettingsClient' appears in /data/app/~~Yu4lRTFSjrQeAB3dMXGmIg==/com.example.app-jK1oINC_g1INHcHHe6qjKA==/base.apk)
                                                                                                    	at com.birjuvachhani.locus.LocusActivity.checkSettings(LocusActivity.kt:273)
                                                                                                    	at com.birjuvachhani.locus.LocusActivity.checkIfLocationSettingsAreEnabled(LocusActivity.kt:243)
                                                                                                    	at com.birjuvachhani.locus.LocusActivity.onPermissionGranted(LocusActivity.kt:178)
                                                                                                    	at com.birjuvachhani.locus.LocusActivity.onRequestPermissionsResult(LocusActivity.kt:165)
                                                                                                    	at android.app.Activity.dispatchRequestPermissionsResult(Activity.java:8794)
                                                                                                    	at android.app.Activity.dispatchActivityResult(Activity.java:8647)
                                                                                                    	at android.app.ActivityThread.deliverResults(ActivityThread.java:5415)
                                                                                                    	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5461)
                                                                                                    	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:54)
                                                                                                    	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2259)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:210)
                                                                                                    	at android.os.Looper.loop(Looper.java:299)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8105)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:556)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1045)
```

Above is the crash log. Below is the code is used

```
 Locus.startLocationUpdates(this) { result ->
            result.location?.let {//Recieved location  }
            result.error?.let { /* Received error! */ }
        }
```

**Comment by larsmathuseck** [comment]
Same issue, a workaround is to downgrade to
`implementation 'com.google.android.gms:play-services-location:20.0.0'`.
Maybe slighly newer versions work too, I did not test.
Crashes with 21.0.1.

**Comment by showrav017** [comment]
Worked well on this combination

`implementation "com.google.android.gms:play-services-location:19.0.1"`
`implementation 'com.github.BirjuVachhani:locus-android:4.1.0'`

**Comment by asadwaheed1** [comment]
> Same issue, a workaround is to downgrade to `implementation 'com.google.android.gms:play-services-location:20.0.0'`. Maybe slighly newer versions work too, I did not test. Crashes with 21.0.1.

Above solution worked very well. Thanks

**Comment by kosratdev** [comment]
@BirjuVachhani Any update regarding the issue?

**Comment by kosratdev** [comment]
@BirjuVachhani, I have fixed the issue and opened a PR. Please check it.

**Comment by BirjuVachhani** [comment]
This should be fixed in the latest version `4.1.1`.
